lib/tools/node-modules/node-modules-builder.ts

```diff
@@ -133,27 +133,165 @@ export class NodeModulesBuilder implements INodeModulesBuilder {
+				// Check if child has been extracted in the parent's node modules, AND THEN in `node_modules`
+				// Slower, but prevents copying wrong versions if multiple of the same module are installed
+				// Will also prevent copying project's devDependency's version if current module depends on another version
+				var modulePath = path.join(currentModulePath, "node_modules", name); // /node_modules/parent/node_modules/<package>
+				var exists = ensureModuleExists(modulePath);
+
+				if (exists) {
+					var dep = addDependency(deps, name, modulePath, depth + 1);
+
+					traverseModule(modulePath, depth + 1, dep);
+				} else {
+					modulePath = path.join(nodeModulesDir, name); // /node_modules/<package>
+					exists = ensureModuleExists(modulePath);
+
+					if(!exists) {
+						return;
+					}
+
+					var dep = addDependency(deps, name, modulePath, 0);
+
+					traverseModule(modulePath, 0, dep);
+				}
+
+			}
+			// module is scoped 
+			else {
+				var scopeSeparatorIndex = name.indexOf('/');
+				var scope = name.substring(0, scopeSeparatorIndex);
+				var moduleName = name.substring(scopeSeparatorIndex + 1, name.length);
+				var scopedModulePath = path.join(nodeModulesDir, scope, moduleName);
+
+				var exists = ensureModuleExists(scopedModulePath);
+
+				if (exists) {
+					var dep = addDependency(deps, name, scopedModulePath, 0);
+					traverseModule(scopedModulePath, depth, dep);
+				}
+				else {
+					scopedModulePath = path.join(currentModulePath, "node_modules", scope, moduleName);
+
+					exists = ensureModuleExists(scopedModulePath);
+
+					if (!exists) {
+						return;
+					}
+
+					var dep = addDependency(deps, name, scopedModulePath, depth + 1);
+					traverseModule(scopedModulePath, depth + 1, dep);
+				}
+			}
+
+			function traverseModule(modulePath: string, depth: number, currentDependency: any) {
+				var packageJsonPath = path.join(modulePath, 'package.json');
+				var packageJsonExists = fs.lstatSync(packageJsonPath).isFile();
+
+				if (packageJsonExists) {
+					var packageJsonContents = require(packageJsonPath);
+
+					if (!!packageJsonContents.nativescript) {
+						// add `nativescript` property, necessary for resolving plugins
+						currentDependency.nativescript = packageJsonContents.nativescript;
+					}
+
+					if (packageJsonContents.dependencies) {
+						Object.keys(packageJsonContents.dependencies).forEach((key) => {
+
+							traverseChild(key, modulePath, depth);
+						});
+					}
+				}
+			}
+
+			function addDependency(deps: any[], name: string, directory: string, depth: number) {
+				var dep: any = {};
+				dep.name = name;
+				dep.directory = directory;
+				dep.depth = depth;
+
+				deps.push(dep);
+
+				return dep;
+			}
+
+			function ensureModuleExists(modulePath: string): boolean {
+				try {
+					var exists = fs.lstatSync(modulePath);
+					return exists.isDirectory();
+				} catch (e) {
+					return false;
+				}
+			}
+		}
+
+		function filterUniqueDirectories(dependencies: any) {
```

Comment:
Noted. I used a string map initially to check if the plugin's already present in the list, but that approach omitted corner cases where 2 plugins with different versions were present. I'll revisit it again